src/javascript/binary/base/passwordmeter.js

```diff
@@ -0,0 +1,179 @@
+    {
+      intScore = (intScore+18);
+      strLog   = strLog + "18 point for length (" + passwd.length + ")\n";
+    }
+
+
+    // LETTERS (Not exactly implemented as dictacted above because of my limited understanding of Regex)
+    if (passwd.match(/[a-z]/))                              // [verified] at least one lower case letter
+    {
+      intScore = (intScore+1);
+      strLog   = strLog + "1 point for at least one lower case char\n";
+    }
+
+    if (passwd.match(/[A-Z]/))                              // [verified] at least one upper case letter
+    {
+      intScore = (intScore+5);
+      strLog   = strLog + "5 points for at least one upper case char\n";
+    }
+
+    // NUMBERS
+    if (passwd.match(/\d+/))                                 // [verified] at least one number
+    {
+      intScore = (intScore+5);
+      strLog   = strLog + "5 points for at least one number\n";
+    }
+
+    if (passwd.match(/(.*[0-9].*[0-9].*[0-9])/))             // [verified] at least three numbers
+    {
+      intScore = (intScore+5);
+      strLog   = strLog + "5 points for at least three numbers\n";
+    }
+
+
+    // SPECIAL CHAR
+    if (passwd.match(/.[!,@,#,$,%,^,&,*,?,_,~]/))            // [verified] at least one special character
+    {
+      intScore = (intScore+5);
+      strLog   = strLog + "5 points for at least one special char\n";
+    }
+
+                   // [verified] at least two special characters
+    if (passwd.match(/(.*[!,@,#,$,%,^,&,*,?,_,~].*[!,@,#,$,%,^,&,*,?,_,~])/))
+    {
+      intScore = (intScore+5);
+      strLog   = strLog + "5 points for at least two special chars\n";
+    }
+
+
+    // COMBOS
+    if (passwd.match(/([a-z].*[A-Z])|([A-Z].*[a-z])/))        // [verified] both upper and lower case
+    {
+      intScore = (intScore+2);
+      strLog   = strLog + "2 combo points for upper and lower letters\n";
+    }
+
+    if (passwd.match(/([a-zA-Z])/) && passwd.match(/([0-9])/)) // [verified] both letters and numbers
+    {
+      intScore = (intScore+2);
+      strLog   = strLog + "2 combo points for letters and numbers\n";
+    }
+
+                  // [verified] letters, numbers, and special characters
+    if (passwd.match(/([a-zA-Z0-9].*[!,@,#,$,%,^,&,*,?,_,~])|([!,@,#,$,%,^,&,*,?,_,~].*[a-zA-Z0-9])/))
+    {
+      intScore = (intScore+2);
+      strLog   = strLog + "2 combo points for letters, numbers and special chars\n";
+    }
+
+
+    if(intScore < 16)
+    {
+       strVerdict = text.localize("Very weak");
+    }
+    else if (intScore > 15 && intScore < 25)
+    {
+       strVerdict = text.localize("Weak");
+    }
+    else if (intScore > 24 && intScore < 35)
+    {
+       strVerdict = text.localize("Mediocre");
+    }
+    else if (intScore > 34 && intScore < 45)
+    {
+       strVerdict = text.localize("Strong");
+    }
+    else
+    {
+       strVerdict = text.localize("Stronger");
+    }
+
+  var array = [intScore, strVerdict];
+  return array;
+}
```

Comment:
zxcvbn is a library created and used by Dropbox - a $10b company.
This one is 10 year old 'quick and dirty password quality meter' - by the words of the authors.
Its calculations give completely incorrect results of the actual entropy.